examples/dynamic.c

```diff
@@ -31,6 +33,10 @@
 #include <unistd.h>
 #include <time.h>
 #include <sys/param.h>
+#include <pmix_config.h>
+#ifdef HAVE_NETDB_H
+#include <netdb.h>
+#endif
```

Comment:
MAXHOSTNAMELEN is defined in netdb.h under Solaris (my commit message was truncated, I will update it tomorrow)
pmix_config.h is where HAVE_NETDB_H is #define'd
netdb.h is likely present under Linux, so removing the #ifdef could be acceptable too